.github/workflows/dependabot-code-gen.yml

```diff
@@ -18,7 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Set up Go 1.x
-      uses: actions/setup-go@d0a58c1c4d2b25278816e339b944508c875f3613 # tag=v3.4.0
+      uses: actions/setup-go@v3 # tag=v3.4.0
```

Comment:
We should remove this comment which is misleading now.

Reply:
I was digging more into the using SHA vs the major version, and came across the below from [Github actions page](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses):
> Using the commit SHA of a released action version is the safest for stability and security.

However, using the major version should not cause any problems. From the official documentation:
> If the action publishes major version tags, you should expect to receive critical fixes and security patches while still retaining compatibility. Note that this behavior is at the discretion of the action's author.


I think the reason why dependabot did not replace the SHA to a version was because it was honoring our initial setup of using the SHA for the commit ID the dependency was release with.

Similar scenario: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3019/files

Reply:
Would you suggest us to switch to using major versions instead for GitHub-actions ?